aria/orchestrator/context/common.py

```diff
@@ -97,19 +99,49 @@ def download_resource(self, destination, path=None):
         Download a blueprint resource from the resource storage
         """
         try:
-            return self.resource.deployment.download(entry_id=self.deployment.id,
-                                                     destination=destination,
-                                                     path=path)
+            self.resource.deployment.download(entry_id=str(self.deployment.id),
+                                              destination=destination,
+                                              path=path)
         except exceptions.StorageError:
-            return self.resource.blueprint.download(entry_id=self.blueprint.id,
-                                                    destination=destination,
-                                                    path=path)
+            self.resource.blueprint.download(entry_id=str(self.blueprint.id),
+                                             destination=destination,
+                                             path=path)
+
+    def download_resource_and_render(self, destination, path=None, variables=None):
+        """
+        Download a blueprint resource from the resource storage render its content as a jinja
+        template using the provided variables. ctx is available to the template without providing it
+        explicitly.
+        """
+        self.download_resource(destination=destination, path=path)
+        with open(destination, 'rb') as f:
```

Comment:
on windows, omitting the `b` will cause newlines to be written as CRLF regardless of how they were originally. So it is definitely required for `wb`. The `rb` is for consistency